.github/workflows/test_comprehensive.yml

```diff
@@ -0,0 +1,165 @@
+# The full test suite, which will be run anytime anything is merged into master.
+# See test_pull_request.yml for the tests that will be run
+name: Comprehensive Tests
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  qa:
+    name: Code Quality
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        task: [flake8, black, isort]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install tox
+      - name: Run task
+        run: tox -e ${{ matrix.task }}
+
+  manifest:
+    name: Check Manifest
+    needs: qa
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install check-manifest
+      - name: Check Manifest
+        run: check-manifest
+
+  test:
+    needs: qa
+    name: ${{ matrix.platform }} py${{ matrix.python }} ${{ matrix.toxenv }} ${{ matrix.MIN_REQ && '(minreq)' }}
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu-latest, windows-latest, macos-latest]
+        python: [3.7, 3.8, 3.9]
+        include:
+          # test big sur on 3.9.0
+          - python: 3.9
+            platform: macos-11.0
+            toxenv: py390-macos-pyqt
+          # test with minimum specified requirements
+          - python: 3.7
+            platform: ubuntu-18.04
+            MIN_REQ: 1
+          # test with --async_only
+          - python: 3.8
+            platform: ubuntu-18.04
+            toxenv: async-py38-linux-pyqt
+          # test with --perfmon-only
+          - python: 3.8
+            platform: ubuntu-18.04
+            toxenv: perfmon-py38-linux-pyqt
```

Comment:
Can we add these back to the `Quick` test too? We've still got an async PR in progress and it would be nice to know as we refactor that we aren't breaking things. We can drop from `Quick` in the future if we feel excessive.

Reply:
Sure can do.  Can someone help me understand what's going on with the perfmon test?  I don't get why we have a dedicated test in the matrix just to [execute a single test function and skip all of the others](https://cirrus-ci.com/task/6286878838095872?command=test_pyqt5#L1655)...

Reply:
Yeah, I think the we should drop `pytest.mark.perfmon`. I think at the time it was thought we might end up with more, but it doesn't make sense to have all the extra machinery for one test. I'll look into making a new PR to drop tonight. I think there are more for `ASYNC`